mobile/lib/modules/trash/providers/trashed_asset.provider.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/home/ui/asset_grid/asset_grid_data_structure.dart';
+import 'package:immich_mobile/modules/trash/services/trash.service.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/models/exif_info.dart';
+import 'package:immich_mobile/shared/providers/asset.provider.dart';
+import 'package:immich_mobile/shared/providers/db.provider.dart';
+import 'package:immich_mobile/shared/providers/user.provider.dart';
+import 'package:isar/isar.dart';
+import 'package:logging/logging.dart';
+
+class TrashNotifier extends StateNotifier<bool> {
+  final Isar _db;
+  final Ref ref;
+  final TrashService _trashService;
+  final log = Logger('TrashNotifier');
+
+  TrashNotifier(
+    this._trashService,
+    this._db,
+    this.ref,
+  ) : super(false);
+
+  Future<void> emptyTrash() async {
+    try {
+      final user = ref.watch(currentUserProvider);
+      if (user == null) {
+        return;
+      }
+      await _trashService.updateTrash(deleteAll: true);
+
+      final assets = await _db.assets
+          .filter()
+          .ownerIdEqualTo(user.isarId)
+          .isTrashedEqualTo(true)
+          .findAll();
+
+      var dbIds = assets.where((a) => a.isRemote).map((e) => e.id).toList();
```

Comment:
you can rewrite this to directly return the remoteId from the Isar DB:

something like `_db.assets.where().remoteIdNotNull().filter().ownerIdEqualTo(user.isarId).isTrahsedEqualTo(true).remoteIdProperty().findAll()`